pkg/apis/machine/validation/machinedeployment.go

```diff
@@ -36,21 +40,45 @@ func canConvertIntOrStringToInt32(val *intstr.IntOrString, replicas int) bool {
 
 func validateUpdateStrategy(spec *machine.MachineDeploymentSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if spec.Strategy.Type != machine.RollingUpdateMachineDeploymentStrategyType && spec.Strategy.Type != machine.RecreateMachineDeploymentStrategyType {
-		allErrs = append(allErrs, field.Required(fldPath.Child("strategy.type"), "Type can either be RollingUpdate or Recreate"))
+
+	supportedStrategies := sets.New(machine.RecreateMachineDeploymentStrategyType, machine.RollingUpdateMachineDeploymentStrategyType, machine.InPlaceUpdateMachineDeploymentStrategyType)
+	if !supportedStrategies.Has(spec.Strategy.Type) {
+		strategies := supportedStrategies.UnsortedList()
+		sort.Slice(strategies, func(i, j int) bool { return strategies[i] < strategies[j] })
```

Comment:
what is the need to sort only to create an error message?

Reply:
The list returned by the map is not ordered, this will make the test flaky, to prevent that we need to sort the list. Also, it looks better user sees the error which is consistently the same.

Reply:
Hi @unmarshall, Humble ping. PTAL.